e2e/test/iothub/telemetry/TelemetrySendE2ETests.cs

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests.Telemetry
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public partial class TelemetrySendE2ETests : E2EMsTestBase
+    {
+        private const int MessageBatchCount = 5;
+        private const int LargeMessageSizeInBytes = 255 * 1024; // The maximum message size for device to cloud messages is 256 KB. We are allowing 1 KB of buffer for message header information etc.
+        private readonly string DevicePrefix = $"{nameof(TelemetrySendE2ETests)}_";
+        private readonly string ModulePrefix = $"{nameof(TelemetrySendE2ETests)}_";
+        private static string ProxyServerAddress = Configuration.IoTHub.ProxyServerAddress;
+
+        [LoggedTestMethod]
+        public async Task Telemetry_DeviceSendSingleTelemetry_Mqtt()
```

Comment:
what do you think about adding tests for amqp? even though we won't advertise support, technically you could still use telemetry over amqp (since we're simply calling SendEvenetAsync underneath.

Reply:
I agree we should. Let me add a work item for that so it doesn't get lost.